python/tk_nuke_writenode/handler.py

```diff
@@ -1023,13 +1023,16 @@ def __set_profile(self, node, profile_name, reset_all_settings=False):
         publish_template = self._app.get_template_by_name(profile["publish_template"])
         proxy_render_template = self._app.get_template_by_name(profile["proxy_render_template"])
         proxy_publish_template = self._app.get_template_by_name(profile["proxy_publish_template"])
-        file_type = profile["file_type"]
         file_settings = profile["settings"]
         tile_color = profile["tile_color"]
         use_node_name = profile["use_node_name"]
         tank_channel = profile["tank_channel"]
         promote_write_knobs = profile.get("promote_write_knobs", [])
 
+        # translate profile extension to nuke extension
+        extension_choices = render_template.keys["extension"].labelled_choices
+        file_type = extension_choices[profile["file_type"]]
+
```

Comment:
Assumes the extensions choices contains a map of extension in file name to extension expected by Nuke (e.g. jpg: jpeg)